test/augmentation/test_base.py

```diff
@@ -165,3 +166,22 @@ def test_gradcheck(self, device, dtype):
         with patch.object(augmentation, "apply_transform", autospec=True) as apply_transform:
             apply_transform.return_value = output
             assert gradcheck(augmentation, ((input, input_param)), raise_exception=True, fast_mode=True)
+
+
+class TestGeometricAugmentationBase2D:
+    @pytest.mark.parametrize("batch_prob", [[True, True], [False, True], [False, False]])
+    def test_autocast(self, batch_prob, device, dtype):
+        if not hasattr(torch, "autocast"):
+            pytest.skip("PyTorch version without autocast support")
+
+        aug = RandomAffine(0.5, (0.1, 0.5), (0.5, 1.5), 1.2, p=1.0)
+        x = torch.rand(len(batch_prob), 5, 10, 7, dtype=dtype, device=device)
+
+        # Explicitly set the batch probability to trigger the different conditions in apply_func()
+        params = aug.forward_parameters(x.shape)
+        params["batch_prob"] = torch.tensor(batch_prob, device=device)
+
+        with torch.autocast(device.type):
+            res = aug(x, params)
```

Comment:
This isn't a valid example? @shijianjian 

is happening the shape mismatch which I commented before: https://github.com/kornia/kornia/actions/runs/3980651646/jobs/6823874452#step:4:9277

Reply:
No. You updated the `batch_prob` parameter by hand. So, it may not be viable parameters. For that workflow, you may refer to

https://github.com/kornia/kornia/blob/9639a45f1e1a33071d53f709d2fb705d41c8f22f/kornia/augmentation/base.py#L154-L156